Remotion/Data/DomainObjects/DomainObject.cs

```diff
@@ -723,7 +729,7 @@ protected internal virtual void OnDeleted (EventArgs args)
 
     private void CheckInitializeEventNotExecuting ()
     {
-      if (_isReferenceInitializeEventExecuting)
+      if (_strategy != null && _strategy.IsInitializing)
         throw new InvalidOperationException("While the OnReferenceInitializing event is executing, this member cannot be used.");
     }
```

Comment:
Since we throw the same exception in the strategy we can consider moving this method to the strategy as public API.

Reply:
This method would be so small afterwards, can't we just throw it away and use the null propagated call afterwards?